File: 523-continuous-subarray-sum/continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        long total=0;
        Set<Integer> s=new HashSet<>();
        int numZeroes=0;
        for(int i:nums){
            if(i%k==0){
                numZeroes++;
                if(numZeroes>1) return true;
                continue;
            }
            numZeroes=0;
            total+=i;
            int rem=(int)(total%k);
            if(total%k==0 || s.contains(rem)) return true;
            s.add(rem);
        }
        return false;
    }
}

File: 523-continuous-subarray-sum/continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        set<int> s;
        long tot=0;
        int numzeroes=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]%k==0){
                numzeroes++;
                if(numzeroes>1) return true;
                continue;
            }
            tot+=nums[i];
            int rem=tot%k;
            if((s.find(rem)!=s.end()||tot%k==0)&& i>0) return true;
            s.insert(rem);
            numzeroes=0;
        }
        return false;
    }
};